### DISCUSSION TITLE
How to change the language?

### DISCUSSION THREAD

**Discussion by DarthMikeSundays** [Orama-Interactive/Pixelorama#432] [Q&A] [answered]
I live in Portugal so the editor defaults to the Portuguese language, but since most tutorials are in English i prefer to use that language. Unfortunately i can't find an option to change that

**Comment by OverloadedOrama** [comment, marked as answer]
@MikeSundaysProgrammer Hello, Pixelorama defaults to the language your device uses, but you can easily change it by going to the Edit menu, Preferences, Language and selecting English.

**Reply by CodexEnt** [reply]
Unfortunately, when Pixelorama changes the language for whatever reason, it's very hard to use Edit->Preferences->Language as it is currently in another language, maybe Turkish or Arabic symbols, so I don't know what entry to click on.
As far as I know, yesterday I was editing a file just fine, then today when I opened it up, the language had changed. I upgraded to the 1.1.7 today as I saw it, but it still does not display the correct language (English)

![snap_2025-12-03_1510](https://github.com/user-attachments/assets/bd1b68b7-2bb1-4cda-846d-2aa24deb7732)

Linux Mint

Would you consider at least storing such a thing within a config.ini file, so if a customer has to override it it would be a simple edit LANG=EN or whatever as a backup?

**Reply by Variable-ind** [reply]
It is already being stored in the config.ini file.

https://github.com/user-attachments/assets/d2235e8e-595a-4545-ab6f-df1add52c5fe

**Reply by CodexEnt** [reply]
For me, the variable is not stored in that file at all, or I would have seen it, as it would have been the first place to check, hence my post. Thanks for the response though. Remember I can't access anything from the program itself, it is all in Arabic/Turkish symbols. We don't want to rely on the  process itself determining the locale, and if so, to at least include an option or other way to override it - if it's wrong (or bugged)

I added it to the config file, Pixelorama opens, does write to the file (as Geany editor tells me) but does not update the program and issue remains the same.

pixelorama-linux-x86_64.zip from itch.io

**Reply by Variable-ind** [reply]
It should be possible by extension 🤔, what locale are you looking for? English?

> so I don't know what entry to click on.

The first entry reverts to the system default language. If the first entry is already selected, then it's likely your default locale setting is messed up in OS setting.

Edit: i see what is the problem...
It is not a language issue at all... IT IS THE FONT

**Reply by CodexEnt** [reply]
Yes, let's see if the font is messed. 
BUT it was good to check the locale settings in System Setings -> Language Settings, as there WAS an issue with it:
It was not applied system-wide for some reason, so we corrected it for our "en_CA" and updated the locale in config.ini

Before and After:
![system-locale-before](https://github.com/user-attachments/assets/945d929e-0cf8-4c46-a3af-cfd6abf598be)
![system-locale-after](https://github.com/user-attachments/assets/99544432-2703-4b8f-be5e-6865777fa6b3)

Please

So it was good to check anyway...

Still..
1) Same issue
2) locale is NOT saved or updated to the config.ini  (which should be in my case "en_CA"), though program IS writing to the file on startup

Maybe the font now you say? Thanks

**Reply by Variable-ind** [reply]
> Edit: i see what is the problem...
It is not a language issue at all... IT IS THE FONT

@CodexEnt Here you go, the solution:
Try each option from the menu shown at 00:20, till the font gets fixed (don't worry if it freezes for a few seconds, it's just loading the font).

The reason i think it's not an RTL language (Turkish, Arabic) is that UI didn't change (The UI itself is different in the RTL mode). Also they are mathematical symbols (integrals etc...)

https://github.com/user-attachments/assets/35bcf919-170e-44cf-aa9b-1a2ff88d1945

**Reply by CodexEnt** [reply]
Yep, that's the ticket! Haha! Hard to even change our font or anything when it is in Klingon symbols!
That was a good troubleshooting that might help others, thanks.
I was using Pixelorama on Windows 10, and started using it on Linux then one day it had this issue. Glad we got it sorted because it's a good program!

**Reply by Variable-ind** [reply]
> Yep, that's the ticket! Haha! Hard to even change our font or anything when it is in Klingon symbols! That was a good troubleshooting that might help others, thanks. I was using Pixelorama on Windows 10, and started using it on Linux then one day it had this issue. Glad we got it sorted because it's a good program!

Glad to help 😁

**Comment by DarthMikeSundays** [comment]
Thanks @OverloadedOrama. Sorry for the obvious question, but I didn't thought, at first, that I would find the preferences under edit so I didn't look there.
